File: voteu-project/src/interfaces/Election.ts
import type { Province, Constituency, Municipality, Affiliation } from '@/interfaces'

export interface Election {
  id: string
  name: string
  proviListLhMap: { [key: number]: Province }
  constiListLhMap: { [key: number]: Constituency }
  muniListLhMap: { [key: string]: Municipality }
  affiListLhMap: { [key: number]: Affiliation }
}

File: java/src/main/java/com/voteU/election/java/models/Election.java
package com.voteU.election.java.models;

import lombok.Getter;
import lombok.Setter;

import java.util.LinkedHashMap;

@Getter
@Setter
public class Election {
    private String id;
    private String name;
    private String date;
    private LinkedHashMap<Integer, Province> proviListLhMap;
    private LinkedHashMap<Integer, Constituency> constiListLhMap;
    private LinkedHashMap<String, Municipality> muniListLhMap;
    private LinkedHashMap<String, PollingStation> poStListLhMap;
    private LinkedHashMap<Integer, Affiliation> affiListLhMap;

    public Election(String id, String name, String date) {
        this.id = id;
        this.name = name;
        this.date = date;
        this.proviListLhMap = new LinkedHashMap<>();
        this.constiListLhMap = new LinkedHashMap<>();
        this.muniListLhMap = new LinkedHashMap<>();
        this.poStListLhMap = new LinkedHashMap<>();
        this.affiListLhMap = new LinkedHashMap<>();
    }
}
